pandas/_libs/internals.pyx

```diff
@@ -74,59 +75,65 @@ cdef class BlockPlacement:
         cdef:
             slice s = self._ensure_has_slice()
             Py_ssize_t start, stop, step, _
+
         if s is not None:
             start, stop, step, _ = slice_get_indices_ex(s)
             return iter(range(start, stop, step))
-        else:
-            return iter(self._as_array)
+
+        return iter(self._as_array)
 
     @property
     def as_slice(self) -> slice:
         cdef:
             slice s = self._ensure_has_slice()
-        if s is None:
-            raise TypeError('Not slice-like')
-        else:
+
+        if s is not None:
             return s
 
+        raise TypeError("Not slice-like")
+
     @property
     def indexer(self):
         cdef:
             slice s = self._ensure_has_slice()
+
         if s is not None:
             return s
-        else:
-            return self._as_array
+
+        return self._as_array
 
     def isin(self, arr):
         from pandas.core.indexes.api import Int64Index
+
         return Int64Index(self.as_array, copy=False).isin(arr)
 
     @property
     def as_array(self):
         cdef:
             Py_ssize_t start, stop, end, _
+
         if not self._has_array:
             start, stop, step, _ = slice_get_indices_ex(self._as_slice)
-            # NOTE: this is the C-optimized equivalent of
-            #  np.arange(start, stop, step, dtype=np.int64)
+            # NOTE:
+            # this is the C-optimized equivalent of
+            # `np.arange(start, stop, step, dtype=np.int64)`
             self._as_array = cnp.PyArray_Arange(start, stop, step, NPY_INT64)
             self._has_array = True
+
         return self._as_array
 
     @property
     def is_slice_like(self) -> bool:
         cdef:
             slice s = self._ensure_has_slice()
+
         return s is not None
 
     def __getitem__(self, loc):
         cdef:
             slice s = self._ensure_has_slice()
-        if s is not None:
-            val = slice_getitem(s, loc)
-        else:
-            val = self._as_array[loc]
+
+        val = slice_getitem(s, loc) if s is not None else self._as_array[loc]
```

Comment:
@jorisvandenbossche This is a very similar change to the change that was mentioned [here](https://github.com/pandas-dev/pandas/pull/32801#discussion_r394307873), would you like me to revert that as well?

Reply:
this is different from the other places where you left this comment, and this is the only one i would ask you to revert.  The way it is currently written, I can look in coverage output to see if both branches are reached; I can't with this PR's version.